### ISSUE TITLE
Output alternates between runs

### ISSUE THREAD

**Issue by MarkusLund** [datalogai/recurrentshop#28]
I created this basic LSTM network while learning RecurrentShop. However the output of the code alternates between two results. **50% of the time the networks trains and outputs something reasonable, while the other half of the runs the network does not train at all.** 

My complete code and outputs from two different runs(take a look at loss).
```
from recurrentshop import RecurrentContainer, LSTMCell
import numpy
from keras.layers import Dense, Input
from keras.models import Model

nb_examples = 10000
x = [[[1, 2, 3, 4]]]*nb_examples + [[[2, 3, 4, 5]]]*nb_examples
y = [10]*nb_examples + [14]*nb_examples

rc = RecurrentContainer(readout=True)

rc.add(LSTMCell(4, input_shape=(4,)))

input = Input(shape=(None, 4))

l = (rc)(input)
output = Dense(1, activation='relu')(l)
model = Model(input=input, output=[input, output])
model.compile(loss='mse', optimizer='adam')
model.fit(x=numpy.asarray(x), y=[numpy.asarray(x), numpy.asarray(y)])

print(model.predict(numpy.asarray([[[1, 2, 3, 4]]])))
```

```
Epoch 1/10
20000/20000 [==============================] - 1s - loss: 111.0854 - input_1_loss: 0.0000e+00 - dense_1_loss: 111.0854     
Epoch 2/10
20000/20000 [==============================] - 0s - loss: 61.1840 - input_1_loss: 0.0000e+00 - dense_1_loss: 61.1840     
Epoch 3/10
20000/20000 [==============================] - 0s - loss: 37.5625 - input_1_loss: 0.0000e+00 - dense_1_loss: 37.5625     
Epoch 4/10
20000/20000 [==============================] - 0s - loss: 22.2517 - input_1_loss: 0.0000e+00 - dense_1_loss: 22.2517     
Epoch 5/10
20000/20000 [==============================] - 0s - loss: 12.7959 - input_1_loss: 0.0000e+00 - dense_1_loss: 12.7959     
Epoch 6/10
20000/20000 [==============================] - 0s - loss: 7.5595 - input_1_loss: 0.0000e+00 - dense_1_loss: 7.5595     
Epoch 7/10
20000/20000 [==============================] - 0s - loss: 5.1144 - input_1_loss: 0.0000e+00 - dense_1_loss: 5.1144     
Epoch 8/10
20000/20000 [==============================] - 0s - loss: 4.0420 - input_1_loss: 0.0000e+00 - dense_1_loss: 4.0420     
Epoch 9/10
20000/20000 [==============================] - 0s - loss: 1.7059 - input_1_loss: 0.0000e+00 - dense_1_loss: 1.7059     
Epoch 10/10
20000/20000 [==============================] - 0s - loss: 0.4829 - input_1_loss: 0.0000e+00 - dense_1_loss: 0.4829     
[array([[[ 1.,  2.,  3.,  4.]]], dtype=float32), array([[ 10.01733208]], dtype=float32)]

Process finished with exit code 0
```

```
Epoch 1/10
20000/20000 [==============================] - 1s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 2/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 3/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 4/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 5/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 6/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 7/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 8/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 9/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
Epoch 10/10
20000/20000 [==============================] - 0s - loss: 148.0000 - input_1_loss: 0.0000e+00 - dense_1_loss: 148.0000     
[array([[[ 1.,  2.,  3.,  4.]]], dtype=float32), array([[ 0.]], dtype=float32)]
```
